### ISSUE TITLE
Sendy node

### ISSUE THREAD

**Issue by danielepolencic** [uasabi/noderosso#49]
_TL;DR: Create a node to send drip campaigns._

Depends on: https://github.com/uasabi/noderosso/issues/52

## Plan

- [ ] Subscribe
- [ ] Unsubscribe
- [ ] Add/remove templates
- [ ] Add/remove drip campaigns

## What it is

Drip campaigns are a sequence of emails that are sent at regular intervals:

1. Welcome email
2. Step 1 (7 days after the previous email)
3. Step 2 (7 days after the previous email)
4. [more emails]
5. Closing email

### Subscribe

```ts
{
  topic: 'SUBSCRIBE.V1',
  payload: {
    email: string, // valid email address
    name?: string
  }
}
```

### Unsubscribe

You should mark the email as unsubscribe. do not remove the record from the context.

```ts
{
  topic: 'UNSUBSCRIBE.V1',
  payload: {
    email: string, // valid email address
  }
}
```

### Flush

Remove all subscribers.

```ts
{
  topic: 'FLUSH.V1',
}
```

### Tick

When the node receives a `TICK.V1`, it checks if any email has to be sent out.

- Filter all active templates that belong to a campaign.
- For each user, check if they are due for the next email in the campaign.
- Group the emails by the template.
- Issue an event for each template with all email addresses for that template.

```ts
{
  topic: 'TICK.V1',
  payload: string | number, // time now, should be parsed with chrono-node
}
```

### Add template

```ts
{
  topic: 'ADD_TEMPLATE.V1',
  payload: {
    id: string
    content: string
  }
}
```

### Remove template

```ts
{
  topic: 'REMOVE_TEMPLATE.V1',
  payload: {
    id: string
  }
}
```

### Add drip campaign

```ts
{
  topic: 'ADD_DRIP_CAMPAIGN.V1',
  payload: {
    id: string
    templateIds: string[]
    interval: humanInterval('7 days') // intervals between emails
  }
}
```

### Remove Drip campaign

```ts
{
  topic: 'REMOVE_DRIP_CAMPAIGN.V1',
  payload: {
    id: string
  }
}
```

### Events

// sendgrid email

**Comment by danielepolencic** [comment]
This is superseded by Sendgrid.
